### PR TITLE
Banlist search

### LINKED ISSUE #380: Feature request: legality filter
Would be real handy to be able to filter cards by set legality, e.g.:
```
t:asset l:standard
```
to return all standard-legal assets

### PR DESCRIPTION
**Pull request by plural**
This adds a new b: operator to search for banlists.  The operator *excludes* any cards banned by that banlist.  For early MWLs without bans, this doesn't do anything.  This (and the rotation operator) will handle latest and current appropriately in a time like today where there is a new *published* rotation/banlist but it isn't yet *active*.

Fixes #380 

This updates the syntax page, syntax tooltip and the advanced search page.  

PAD tap properly excluded because it's banned, but in the legal card pool.
search-results-pad-tap-banned

Updated Search Syntax Page
search-syntax-page

Advanced Search Page
advanced-search

### CODE REVIEW

File: src/AppBundle/Service/CardsData.php [hunk 1]
```diff
@@ -583,12 +583,30 @@ public function get_search_rows(array $conditions, string $sortorder, string $lo
...
+                    if ($condition[0] == "active") {
+                        $mwl = $this->entityManager->getRepository(Mwl::class)->findOneBy(['active' => 1], ["dateStart" => "DESC"]);
+                    } elseif ($condition[0] == "latest") {
+                        $mwl = $this->entityManager->getRepository(Mwl::class)->findOneBy([], ["dateStart" => "DESC"]);
+                    } else {
+                        $mwl = $this->entityManager->getRepository(Mwl::class)->findOneBy(['code' => $condition[0]], ["dateStart" => "DESC"]);
+                    }
+                    if ($mwl) {
+                        // Exclude any cards banned by this banlist.
+                        $clauses[] = "(c.id NOT IN (SELECT mc.card_id FROM AppBundle:MwlCard mc WHERE mc.mwl_id = ?$i ) )";
```
Comment (NoahTheDuke) [inline src/AppBundle/Service/CardsData.php:597]: Spacing around parentheses

**Review by NoahTheDuke — APPROVED** [review]

**Comment by NoahTheDuke** [comment]
Make sure you and @gereons figure out how this interacts with his rotations PR.

**Comment by gereons** [comment]
I _think_ the change required would be pretty trivial, namely where the search currently does

    $clauses[] = "(y.code in (" . implode(", ", $placeholders) . "))";

we would have to negate the query, i.e. use `not in`

**Comment by plural** [comment]
let's make that update in your PR, @gereons .

We'll want to coordinate releasing this with the rotation change so we have one bit of downtime for the db update/migration.

**Pull request merged**